src/test/java/org/cactoos/scalar/AvgOfTest.java

```diff
@@ -51,213 +51,313 @@ public void withEmptyCollection() {
-                new ListOf<>(Long.MAX_VALUE, Long.MAX_VALUE)
-                    .toArray(new Long[2])
-                        ).longValue(),
-                Matchers.equalTo(Long.MAX_VALUE)
+                Float.MAX_VALUE, Float.MAX_VALUE
+            ).floatValue(),
+            Matchers.equalTo(Float.MAX_VALUE)
         );
+    }
+
+    @Test
+    public void withFloatCollectionMinValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(Float.MAX_VALUE, Float.MAX_VALUE)
-                    .toArray(new Float[2])
-                        ).floatValue(),
-                Matchers.equalTo(Float.MAX_VALUE)
+                Float.MIN_VALUE, Float.MIN_VALUE
+            ).floatValue(),
+            Matchers.equalTo(Float.MIN_VALUE)
         );
     }
 
     @Test
-    public void withMinValueCollections() {
+    public void withIntScalarsIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(Double.MIN_VALUE, Double.MIN_VALUE)
-                    .toArray(new Double[2])
-                        ).doubleValue(),
-                Matchers.equalTo(Double.MIN_VALUE)
+                () -> 1, () -> 2, () -> 10
+            ).intValue(),
+            Matchers.equalTo(4)
         );
+    }
+
+    @Test
+    public void withLongScalarsIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(Integer.MIN_VALUE, Integer.MIN_VALUE)
-                    .toArray(new Integer[2])
-                        ).intValue(),
-                Matchers.equalTo(Integer.MIN_VALUE)
+                () -> 1L, () -> 2L, () -> 10L
+            ).intValue(),
+            Matchers.equalTo(4)
         );
+    }
+
+    @Test
+    public void withFloatScalarsIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(Long.MIN_VALUE, Long.MIN_VALUE)
-                    .toArray(new Long[2])
-                        ).longValue(),
-                Matchers.equalTo(Long.MIN_VALUE)
+                () -> 1f, () -> 2f, () -> 10f
+            ).longValue(),
+            Matchers.equalTo(4L)
         );
+    }
+
+    @Test
+    public void withDoubleScalarsIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(Float.MIN_VALUE, Float.MIN_VALUE)
-                    .toArray(new Float[2])
-                        ).floatValue(),
-                Matchers.equalTo(Float.MIN_VALUE)
+                () -> 1.0d, () -> 2.0d, () -> 10.0d
+            ).intValue(),
+            Matchers.equalTo(4)
         );
     }
 
     @Test
-    public void withEvenOddValues() {
-        final Float[] array = new ListOf<>(12345.0f, -12345.0f)
-            .toArray(new Float[2]);
-        final int expected = 0;
-        MatcherAssert.assertThat(
-            new AvgOf(array).intValue(), Matchers.equalTo(expected)
-        );
+    public void withIterableOfScalars() {
```

Comment:
@allanws Please, add more tests, including using composition with other `Scalar` classes, for example:
```
new AvgOf(
  new MaxOf(2, 3),
  new MinOf(5, 9),
  new SumOf(1, 2, 3)
).intValue()
```
And don't forget the `Ternary` class too.